### PR TITLE
DOC-8738: Warn users that generating the debug zip can degrade the cluster

### PR DESCRIPTION
**Pull request by florence-crl**
Addresses DOC-8738

(1) Updated cockroach debug zip page with cluster impact warning and recommendation to filter by time range.

### CODE REVIEW

**Review by kevinkokomani — APPROVED** [review]
LGTM

File: src/current/v23.1/cockroach-debug-zip.md [hunk 1]
```diff
@@ -18,16 +18,22 @@ The files produced by `cockroach debug zip` can contain highly [sensitive, perso
 
 ### Use cases
 
+{{site.data.alerts.callout_danger}}
+ `cockroach debug zip` is an expensive operation and impacts cluster performance.
+
+The command is to be used only as an emergency measure under the guidance of Cockroach Labs.
```
Comment (gemma-shay) [inline src/current/v23.1/cockroach-debug-zip.md:24]:
```suggestion
Only use this command as an emergency measure under the guidance of Cockroach Labs.
```

File: src/current/v23.2/cockroach-debug-zip.md [hunk 2]
```diff
@@ -18,16 +18,22 @@ The files produced by `cockroach debug zip` can contain highly [sensitive, perso
 
 ### Use cases
 
+{{site.data.alerts.callout_danger}}
+ `cockroach debug zip` is an expensive operation and impacts cluster performance.
+
+The command is to be used only as an emergency measure under the guidance of Cockroach Labs.
```
Comment (gemma-shay) [inline src/current/v23.2/cockroach-debug-zip.md:24]:
```suggestion
Only use this command as an emergency measure under the guidance of Cockroach Labs.
```

**Review by gemma-shay — APPROVED** [review]

File: src/current/v23.1/cockroach-debug-zip.md (hunk 1 above)
Comment (florence-crl) [reply src/current/v23.1/cockroach-debug-zip.md]: updated

**Review by kevin-v-ngo — COMMENTED** [review]
Hi @florence-crl, do you think we can include guidance and document this new flag? We have a few threads and issues related to the range info filter.

Doc issue: https://github.com/cockroachdb/docs/issues/16919
PR: https://github.com/cockroachdb/cockroach/pull/102289

**Comment by florence-crl** [comment]
Before I can document https://github.com/cockroachdb/docs/issues/16919, I need to clarify what is the default value. For context, see [this slack thread](https://cockroachlabs.slack.com/archives/C01CNRP6TSN/p1696863744025449).

**Comment by florence-crl** [comment]
Clarification: https://github.com/cockroachdb/cockroach/pull/104133 changed the default to `true` and concat-ed the range info into 1 `ranges.json` file.

**Comment by florence-crl** [comment]
@dt, PTAL, I included your recommendation under the `--include-range-info` flag description.

File: src/current/v23.1/cockroach-debug-zip.md [hunk 3]
```diff
@@ -105,6 +111,7 @@ Flag | Description
...
 `--include-files` | [Files](#files) to include in the generated `.zip`. This can be used to limit the size of the generated `.zip`, and affects logs, heap profiles, goroutine dumps, and/or CPU profiles. The files are specified as a comma-separated list of [glob patterns](https://wikipedia.org/wiki/Glob_(programming)). For example:<br /><br />`--include-files=*.pprof`<br /><br />Note that this flag is applied _before_ `--exclude-files`. Use [`cockroach debug list-files`]({% link {{ page.version.version }}/cockroach-debug-list-files.md %}) with this flag to see a list of files that will be contained in the `.zip`.
+`--include-range-info` | Include information about ranges in <code>nodes/{node ID}/ranges.json</code> files, a single file per node.<br /><br />Only set this flag to `false` to exclude range information when all the following conditions are met:<ul><li>You are directed by Cockroach Labs support.</li><li>The issue under investigation is definitely not related to kv, for example, an error from a schema change about versions or flags or unsupported.</li><li>The cluster is large and you need to minimize the size of the `.zip` file.</ul>**Default:** true
```
Comment (dt) [inline src/current/v23.1/cockroach-debug-zip.md:114]: I think for user-facing docs, I'd say be more specific than "related to kv". I might go with something like:

Include one file per node with information about the KV ranges stored on that node, in <code>nodes/{node ID}/ranges.json</code>. This information can be vital when debugging issues that involve the KV storage layer, such as data placement, load balancing, performance or other behaviors. In certain situations, on large clusters with large numbers of ranges, this file can be omitted if and only if the issue being investigated is already known to be in another layer of the system (for example an error message about an unsupported feature or incompatible value in a SQL schema change or statement). Note however many higher-level issues are ultimately related to the underlying KV storage layer described by this file so only set this to false if directed to do so by Cockroach Labs support. *Default:** true

Comment (florence-crl) [reply src/current/v23.1/cockroach-debug-zip.md]: updated

**Review by dt — APPROVED** [review]

**Review by kevin-v-ngo — APPROVED** [review]
Nice! Looks good Florence. Thanks for the updates.

**Pull request merged**